### PR TITLE
Type annotate bandersnatch plugins

### PR DESCRIPTION
**Pull request by ichard26**
Related: #15.

Type annotate the bandersnatch plugins.

@techalchemy, it would be nice if you could review this PR since you know these plugins the best. (Only if you have the time though)

### CODE REVIEW

**Comment by ichard26** [comment]
I also learned `# type: ignore[override]` today. I was a little annoyed with `Signature incompatible with Super type` warnings. Also, trust me, the only reason why it looks *legit* is because of the amount of time I spent on it.

**Comment by cooperlees** [comment]
And I appreciate it. As someone who’s typed a lot of old code, I appreciate the difficulties. Typing code changes how you code moving forward. For the better I believe :)

File: src/bandersnatch_filter_plugins/prerelease_name.py [hunk 1]
```diff
@@ -31,13 +32,13 @@ def initialize_plugin(self):
...
 
-    def filter(self, metadata):
+    def filter(self, metadata: Dict) -> bool:
         """
         Remove all release versions that match any of the specified patterns.
         """
         releases = metadata["releases"]
         for version in list(releases.keys()):
-            if any(pattern.match(version) for pattern in self.patterns):
+            if any(pattern.match(version) for pattern in self.patterns):  # type: ignore
```
Comment (cooperlees) [inline src/bandersnatch_filter_plugins/prerelease_name.py:41]: Will be interested to see what the error is here.

Comment (ichard26) [reply src/bandersnatch_filter_plugins/prerelease_name.py:41]:
```
prerelease_name.py:41: error: Item "None" of "Optional[List[Pattern[str]]]" has no attribute "__iter__" (not iterable)
```

Comment (cooperlees) [reply src/bandersnatch_filter_plugins/prerelease_name.py:41]: Ahh - makes sense.

Yeah, here we just need to add another check for a None object and handle it. I’ll go back and look at type ignores before closing the issue and see if we can get rid of a few more.

**Pull request merged**